src/main/scala/net/psforever/actors/session/support/SessionData.scala

```diff
@@ -1640,92 +1403,392 @@ class SessionData(
+
+          case e: Equipment =>
+            (
+              player.Holsters().zipWithIndex.flatMap { case (o, i) =>
+                o.Equipment.flatMap { item => Some((player, InventoryItem(item, i))) }
+              }.toList ++
+                player.Inventory.Items.map { o => (player, o) } ++
+                player.FreeHand.Equipment.flatMap { item => Some((player, InventoryItem(item, Player.FreeHandSlot))) }.toList ++
+                (validObject(player.VehicleSeated) match {
+                  case Some(v: Vehicle) => v.Trunk.Items.map { o => (v, o) }
+                  case _ => Nil
+                })
+              )
+              .find { case (_, InventoryItem(o, _)) => o eq e } match {
+              case Some((p: Player, InventoryItem(obj, index))) if !obj.HasGUID =>
+                p.Slot(index).Equipment = None
+                attemptRecoveryFromNoGuidExceptionAsPlayer(p, nge)
+              case Some((p: Player, _)) =>
+                attemptRecoveryFromNoGuidExceptionAsPlayer(p, nge)
+
+              case Some((v: Vehicle, InventoryItem(obj, index))) if !obj.HasGUID && v.PassengerInSeat(player).contains(0) =>
+                v.Slot(index).Equipment = None
+                attemptRecoveryFromNoGuidExceptionAsPlayer(player, nge)
+                SupervisorStrategy.resume
+              case Some((v: Vehicle, InventoryItem(obj, index))) if !obj.HasGUID =>
+                v.Slot(index).Equipment = None
+                SupervisorStrategy.resume
+              case Some((v: Vehicle, _)) if v.PassengerInSeat(player).contains(0) =>
+                attemptRecoveryFromNoGuidExceptionAsPlayer(player, nge)
+                SupervisorStrategy.resume
+              case Some((_: Vehicle, _)) =>
+                SupervisorStrategy.resume
+
+              case _ =>
+                //did not discover or resolve the situation
+                writeLogExceptionAndStop(nge)
+            }
+          case _ =>
+            SupervisorStrategy.resume
+        }
+
+      case ide: InventoryDisarrayException =>
+        attemptRecoveryFromInventoryDisarrayException(ide.inventory)
+        //re-evaluate results
+        if (ide.inventory.ElementsOnGridMatchList() > 0) {
+          writeLogExceptionAndStop(ide)
+        } else {
+          SupervisorStrategy.resume
+        }
+
+      case e =>
+        writeLogExceptionAndStop(e)
+    }
+  }
+
+  def attemptRecoveryFromNoGuidExceptionAsVehicle(v: Vehicle, e: Throwable): SupervisorStrategy.Directive = {
+    val entry = v.Seats.find { case (_, s) => s.occupants.contains(player) }
+    entry match {
+      case Some(_) =>
+        player.VehicleSeated = None
+        v.Seats(0).unmount(player)
+        player.Position = v.Position
+        player.Orientation = v.Orientation
+        zoning.interstellarFerry = None
+        zoning.interstellarFerryTopLevelGUID = None
+        attemptRecoveryFromNoGuidExceptionAsPlayer(player, e)
+      case None =>
+        writeLogException(e)
+    }
```

Comment:
that can be `collect`

Reply:
Best or worst case, `collect` goes over every entry.  `find` stops on the first positive result.

Reply:
ok